src/stats.c

```diff
@@ -1073,10 +1073,13 @@ static UInt ExecIntrStat(Stat stat)
     /* and now for something completely different                          */
 #ifdef USE_GASMAN
     if ( SyStorOverrun != 0 ) {
-      SyStorOverrun = 0; /* reset */
-      ErrorReturnVoid(
-  "reached the pre-set memory limit\n(change it with the -o command line option)",
-        0, 0, "you can 'return;'" );
+        Int SyStorValue = SyStorOverrun;
+        SyStorOverrun = 0; /* reset */
+        if (SyStorValue != SY_STOR_OVERRUN_HANDLED) {
```

Comment:
So, here is the general picture (as I see it)

At present, IntrExecStat always prints something, either "out of memory" (if SyStorOverrun is non-zero), or "ctrl+c pressed" (if SyStorOverrun is 0).

It's hard to avoid having SyStorOverrun set before we realise our allocation request was too big, so the interrupt is set. At this point we have two choices:

1) Unset to interrupt before it is ever called
2) Let it be called, but don't print anything.

(1) seems fiddly, and I'm worried about "unsetting" it incorrectly, so I thought doing (2) seemed like a better plan.

ExecIntrStat is more the function I'm interested in, because if I try printing an Error, that error ends up being interuppted by ExecIntrStat as soon as it starts.

However, now I've written this all down, I think I have a better way of handling all this :)